akka-stream/src/main/scala/akka/stream/ActorMaterializer.scala

```diff
@@ -129,9 +130,9 @@ object ActorMaterializer {
   /**
    * INTERNAL API
    */
-  private[akka] def downcast(materializer: Materializer): ActorMaterializer =
-    materializer match { //FIXME this method is going to cause trouble for other Materializer implementations
-      case m: ActorMaterializer ⇒ m
+  def downcast(materializer: Materializer): ExtendedActorMaterializer =
```

Comment:
Hi @patriknw 
The problem is here 
```
private[akka] final class ActorRefSink[In](ref: ActorRef, onCompleteMessage: Any,
                                           val attributes: Attributes,
                                           shape: SinkShape[In]) extends SinkModule[In, NotUsed](shape) {

  override def create(context: MaterializationContext) = {
    val actorMaterializer = ActorMaterializer.downcast(context.materializer)
    val effectiveSettings = actorMaterializer.effectiveSettings(context.effectiveAttributes)
    val subscriberRef = actorMaterializer.actorOf(context,
      ActorRefSinkActor.props(ref, effectiveSettings.maxInputBufferSize, onCompleteMessage))
    (akka.stream.actor.ActorSubscriber[In](subscriberRef), NotUsed)
  }

  override protected def newInstance(shape: SinkShape[In]): SinkModule[In, NotUsed] =
    new ActorRefSink[In](ref, onCompleteMessage, attributes, shape)
  override def withAttributes(attr: Attributes): AtomicModule =
    new ActorRefSink[In](ref, onCompleteMessage, attr, amendShape(attr))
}
```
If we downcast to ActorMaterializer then the call to actorOf is private[akka] in the create call above.